spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/functions/DaysFunction.java

```diff
@@ -31,6 +31,8 @@
  * A Spark function implementation for the Iceberg day transform.
  *
  * <p>Example usage: {@code SELECT system.days('source_col')}.
+ *
+ * <p>Alternate form: {@code SELECT system.day('source_col')}.
  */
 public class DaysFunction extends UnaryUnboundFunction {
```

Comment:
@ebyhr if I understand you correctly, you also suggest introducing a constructor that takes a `String name` and then have `name()` returns this name, right?
So `new DaysFunction("day")` would return "day" when its `name()` is called, while `new DaysFunction("days")` would return "days".
I think that introduces more complexity than it's worth.

Reply:
We don't want the function to be constructed with an arbitrary name. We really just want to name to be either "day" or "days" in this case. So we can have a constructor with a boolean.

Reply:
I'm also not keen about having an arguement to this constructor. Why Shouldn't I have DayFunction and DaysFunction. Where DayFunction just extends DaysFunction but overrides the name?